charts/metrics-server/values.yaml

```diff
@@ -127,20 +127,14 @@ addonResizer:
   enabled: false
   image:
     repository: registry.k8s.io/autoscaling/addon-resizer
-    tag: 1.8.14
-  resources:
-    limits:
-      cpu: 40m
-      memory: 25Mi
-    requests:
-      cpu: 40m
-      memory: 25Mi
+    tag: 1.8.19
+  resources: {}
   nanny:
```

Comment:
Don't understand this change. This is the source of truth for nanny resources https://github.com/kubernetes-sigs/metrics-server/blob/master/manifests/components/autoscale/patch.yaml

Reply:
The values in that manifest (and the current Helm chart) caused OOM outages for Metrics Server on a 3 node cluster when I tested this functionality, however the [README](https://github.com/kubernetes-sigs/metrics-server#scaling) values work as expected.

Reply:
Then we should update all of them.

Reply:
I think we are missing `--minClusterSize` flag that is provided in Kubernetes addons (used for testing Kubernetes upstream).  https://github.com/kubernetes/kubernetes/blob/d4fde1e92a83cb533ae63b3abe9d49f08efb7a2f/cluster/addons/metrics-server/metrics-server-deployment.yaml#L118C15-L118C29

It sets `--minClusterSize` to 16 https://github.com/kubernetes/kubernetes/blob/d4fde1e92a83cb533ae63b3abe9d49f08efb7a2f/cluster/gce/gci/configure-helper.sh#L2838

For your 3 node setup it would mean that MS would get 36m cores and 45MiB memory. I think we should improve this. 

/cc @dgrisonnet

Reply:
@serathius the chart has the min cluster size set to 100, as per the README. Using the configuration in this PR (based on the README) to override the default chart values we were able to go from 3 nodes to well past 100.

Reply:
Ok, that's strange. Can you say how much cpu and memory requests your setup is setting?

Reply:
I think we interpreting the `--minClusterSize` differently, let me double check.

Reply:
> For the default nanny behaviour it's configured exactly as the README describes; up to 100 nodes the MS container will have 100m & 200Mi, after that each additional cluster node will add 1m & 2Mi using a threshold of 5.

This is incorrect.

https://github.com/kubernetes/autoscaler/blob/51b4f52c3835759d509a527aaa6fd64324a1a005/addon-resizer/nanny/main/pod_nanny.go#L63

The resource equation is (+/-) exponential estimator that just buckets number of nodes 
```
base + extra * max(numberOfNodes, minClusterSize)
```

For example CPU for 3 node cluster with base 20m, and extra 1m will be: `20 + 1 * max(16,3) = 36m` and the memory will be `15 + 2 * max(16,3)` = 47Mi`.

Can we agree that changing the base to `100m` cores and `200Mi` is bad? If the current resources set in https://github.com/kubernetes-sigs/metrics-server/blob/master/manifests/components/autoscale/patch.yaml are not enough we need to update the estimates.

If you want to set the configuration exactly as the README, (maybe we should do just that) they would be:
```
- --cpu=0Mi
- --extra-cpu=1m
- --memory=0Mi
- --extra-memory=2Mi
- --minClusterSize=100
```

Reply:
PTAL https://github.com/kubernetes-sigs/metrics-server/pull/1304

Reply:
@serathius I'll reply on #1304 as we can then base this PR on the outcome.

Reply:
Done